packages/auth/graphql/resolvers/_mutations/addUserToRole.js

```diff
@@ -4,20 +4,28 @@ const transformUser = require('../../../lib/transformUser')
 const {
   ensureUserHasRole,
   userHasRole,
-  addUserToRole
+  addUserToRole,
+  isRoleClaimableByMe
 } = require('../../../lib/Roles')
-const logger = console
 
-module.exports = async (_, args, { pgdb, req, signInHooks }) => {
-  ensureUserHasRole(req.user, 'admin')
-  const { userId, role } = args
-  const user = await pgdb.public.users.findOne({id: userId})
+module.exports = async (_, args, { pgdb, signInHooks, user: me }) => {
+  const {
+    userId = me && me.id,
+    role
+  } = args
 
+  if (!(me && me.id === userId && isRoleClaimableByMe(role, me))) {
+    ensureUserHasRole(me, 'admin')
+  }
+
+  const user = await pgdb.public.users.findOne({ id: userId })
   if (!user) {
-    logger.error('user not found', { req: req._log() })
     throw new Error(t('api/users/404'))
   }
 
-  const returnedUser = userHasRole(user, role) ? user : (await addUserToRole(userId, role, pgdb))
+  const returnedUser = userHasRole(user, role)
+    ? user
+    : (await addUserToRole(userId, role, pgdb))
+
```

Comment:
If we made it this far, a `user` object was fetch from database, originating either from `userId` argument or `me` object. This, however, won't bother us here and we just stick to `user.id`.

```suggestion
  if (userHasRole(user, role) {
    return user
  }
  
  return addUserToRole(user.id, role, pgdb)
```